packages/react-components/react-carousel-preview/library/src/components/CarouselAnnouncer/CarouselAnnouncer.types.ts

```diff
@@ -0,0 +1,38 @@
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+
+export type CarouselAnnouncerSlots = {
+  root: Slot<'div'>;
+};
+
+export type AnnouncerIndexRenderFunction = (index: number, totalSlides: number, slideGroupList: number[][]) => string;
+/**
+ * CarouselAnnouncer Props
+ */
+export type CarouselAnnouncerProps = Omit<ComponentProps<Partial<CarouselAnnouncerSlots>>, 'children'> & {
+  children: AnnouncerIndexRenderFunction;
+};
+
+/**
+ * State used in rendering CarouselAnnouncer
+ */
+export type CarouselAnnouncerState = ComponentState<CarouselAnnouncerSlots> & {
+  /**
+   * The function that will render nav items based on total slides and their index.
+   */
+  renderAnnouncerChild: AnnouncerIndexRenderFunction;
+
+  /**
+   * The total number of slides passed in from carousel context.
+   */
+  totalSlides: number;
```

Comment:
curious - could/is this info stored somewhere on the carousel context?

Reply:
Most of the carousel logic relies on a pub/sub of carousel events emitted, anyone can subscribe to this within the Carousel container so it's somewhat similar to context - I think we'd want to keep this logic separate as we intend for Carousel to have as few state updates as possible and setting a totalSlides context prop would drive a re-render of entire carousel.